### PR TITLE
Bugfix/Optimize queries for article search

### PR DESCRIPTION
**Pull request by Andriy1328**
1. Optimized queries for article search

### CODE REVIEW

**Review by Pakholchuk-Oleh — APPROVED** [review]
lgtm

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -1,14 +1,17 @@
 # frozen_string_literal: true
 
 class ArticlesController < ApplicationController
+  # rubocop:disable Metrics/AbcSize
   def index
     @articles = if !params[:query] || params[:query].empty?
-                  Article.published.in_language(extract_locale).sorted_desc
+                  Article.includes(:article_tags, :tags).published.in_language(extract_locale).sorted_desc
```
Comment (mmrshk) [inline app/controllers/articles_controller.rb:7]: Move this `Article.includes(:article_tags, :tags).published.in_language(extract_locale)` to another method for examples `def articles`

**Review by mmrshk — CHANGES_REQUESTED** [review]
Fix

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (mmrshk) [reply app/controllers/articles_controller.rb:7]: And after that check do you really need rubocop disable

Comment (olcplif) [reply app/controllers/articles_controller.rb:7]: And you can reuse the sorting for search result

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -16,6 +16,11 @@ def show
     @article = Article.find(params[:id])
   end
 
+  def search_result
```
Comment (mmrshk) [inline app/controllers/articles_controller.rb:19]: memoization??

**Pull request merged**